metaswitch/common/utils.py

```diff
@@ -399,3 +400,22 @@ def sigusr1_handler(sig, stack):
         stack_dump = "Caught SIGUSR1\n" + "".join(traceback.format_stack(stack))
         write_core_file(process_name, stack_dump)
     signal.signal(signal.SIGUSR1, sigusr1_handler)
+
+def write_pid_file(filename):
+    """ Attempts to write a pidfile. If that pidfile is currently
+    locked, returns -1 - the caller should exit at that point."""
+    pid = os.getpid()
+    pidfile = open(filename, "w")
+    try:
+        flock(pidfile, LOCK_EX | LOCK_NB)
+        _log.info("Acquired exclusive lock on %s", filename)
+    except IOError:
+        _log.error("Lock on %s is held by another process", filename)
+        return None
```

Comment:
Will do - that also means if you forget to do the check, subsequent processes exit (slightly less cleanly) rather than carrying blindly on.